### ISSUE TITLE
Creating a formation with quantity=0 is not respected (but updating to 0 is respected)

### ISSUE THREAD

**Issue by daweimau** [heroku/terraform-provider-heroku#351]
### Terraform, provider versions
```
Terraform v1.3.5
on darwin_arm64
+ provider registry.terraform.io/hashicorp/aws v4.40.0
+ provider registry.terraform.io/heroku/heroku v5.1.6
```

### Affected Resource(s)
- `heroku_formation`

### Terraform Configuration Files
```hcl-terraform
terraform {
  backend "local" {}
  required_providers {
    heroku = {
      source  = "heroku/heroku"
      version = "~> 5.0"
    }
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.0"
    }
  }
}
variable "project_name" {
  description = "Project name which will be used to prefix some resource names"
  default     = "myproject"
}
variable "heroku_team_name" {
  description = "The name of the heroku team which will have ownership of this project"
  default     = "myproject"
}
data "heroku_team" "this" {
  name = var.heroku_team_name
}


# Pipeline
resource "heroku_pipeline" "this" {
  name = "${var.project_name}-pipeline"
  owner {
    id   = data.heroku_team.this.id
    type = "team"
  }
}
## scraper
resource "heroku_app" "scraper-staging" {
  name       = "${var.project_name}-scraper-staging"
  region     = "us"
  acm        = true
  buildpacks = ["heroku/nodejs"]
  organization {
    name = data.heroku_team.this.name
  }
}
# Place into the pipeline
resource "heroku_pipeline_coupling" "scraper-staging" {
  app_id   = heroku_app.scraper-staging.id
  pipeline = heroku_pipeline.this.id
  stage    = "staging"
}
# Trigger a build
resource "heroku_build" "scraper-staging" {
  app_id = heroku_app.scraper-staging.id
  depends_on = [
    heroku_app.scraper-staging
  ]
  source { path = "home-scraper/scraper" }
}
# Scale off the auto created dyno
resource "heroku_formation" "scraper-staging" {
  app_id     = heroku_app.scraper-staging.id
  type       = "web"
  quantity   = 0
  size       = "hobby"
  depends_on = [heroku_app.scraper-staging, heroku_build.scraper-staging]
}
```

### Debug Output

https://gist.github.com/daweimau/a9d27393e3ebf3733c4fb62a467b4ecd

### Expected Behavior
The formation should be scaled to 0 (in other words, the dyno should be off). Expected formation is `web=0:Hobby`. 

### Actual Behavior
The formation is scaled to 1 (the dyno is enabled and billing). Actual formation is `web=1:Hobby`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Run `terraform apply`. You will need an authenticated Heroku account and to replace the `heroku_team_name` variable with a Heroku team 
2. Run `heroku scale -a homebody-scraper-staging` . The output will be: `web=1:Hobby`. 

### Important Factoids

- The very tail of the debug output seems extremely relevant:
```
2022-11-22T23:19:51.312+1100 [WARN]  Provider "provider[\"registry.terraform.io/heroku/heroku\"]" produced an unexpected new value for heroku_formation.scraper-staging, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .quantity: was cty.NumberIntVal(0), but now cty.NumberIntVal(1)
```

- Creating a formation with any other valid `quantity` value is respected. For example, `quantity=2` with `size="standardx1"` will work as expected.

- Once the formation exists, the following steps can forcefully *update* the dyno into a `quantity=0` state:
1. Update the terraform plan to `quantity=1`.  Run `terraform apply`.  (Resulting state: `web=1:Hobby`)
2. Revert the plan to `quantity=0`. Run `terraform apply` again. (Resulting state: `web=0:Hobby`)

### References
None that I can find.

**Comment by mars** [comment]
Looking into this, I found that on create, the formation will not set `0` for `quantity`:
https://github.com/heroku/terraform-provider-heroku/blob/v5.1.6/heroku/resource_heroku_formation.go#L110

…because that `GetOk` function will not return the type's zero-value, `0` in this integer case:
https://github.com/heroku/terraform-provider-heroku/blob/v5.1.6/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource_data.go#L88-L93

On update, this is not an issue, because it only checks for change in value:
https://github.com/heroku/terraform-provider-heroku/blob/v5.1.6/heroku/resource_heroku_formation.go#L147

So, this will require a fix to the `heroku_formation` code to allow setting `quantity=0` on create.

**Comment by mars** [comment]
This should be fixed in [v5.1.7](https://registry.terraform.io/providers/heroku/heroku/5.1.7).
